### DISCUSSION TITLE
Edge Classification without node features

### DISCUSSION THREAD

**Discussion by QianChang-page** [pyg-team/pytorch_geometric#6800] [Q&A] [answered]
Thanks for your interest in my question.
I have a dataset including edge connection matrix (Data.edge_index) , edge features (data.edge_attr) and edge label (data.y). 
I'm clueless on how to set up my model to achieve this Edge Classification.

**Comment by rusty1s** [comment, marked as answer]
In this case, you can either learn a shallow embedding matrix via `torch.nn.Embedding` to act as node features, or create a one-vector that acts as a dummy node feature.

**Reply by QianChang-page** [reply]
Thanks a lot, I  treat the actual edges as “nodes” to be linked, and each “node” is the origin edge,the edge features matrix can act as node features .This turned out to be OK for me.

**Reply by rusty1s** [reply]
Yes, this works as well, in which case you are working on the line graph representation.
